packages/wasm/crate/src/planner.rs

```diff
@@ -480,3 +483,25 @@ pub async fn plan_transaction(
 
     Ok(serde_wasm_bindgen::to_value(&plan)?)
 }
+
+/// When planning an undelegate action, there may not be metadata yet in the
+/// IndexedDB database for the unbonding token that the transaction will output.
+/// That's because unbonding tokens are tied to a specific height. If unbonding
+/// tokens for a given validator and a given height don't exist yet, we'll
+/// generate them here and save them to the database, so that they can render
+/// correctly in the transaction approval dialog.
+async fn save_unbonding_token_metadata_if_needed(
+    undelegate: &Undelegate,
+    storage: &IndexedDBStorage,
+) -> WasmResult<()> {
+    let metadata = undelegate.unbonding_token().denom();
+
+    if storage.get_asset(&metadata.id()).await?.is_none() {
+        let metadata_proto = metadata.to_proto();
+        let customized_metadata_proto = customize_symbol(metadata_proto);
+        let customized_metadata = Metadata::try_from(customized_metadata_proto)?;
```

Comment:
It seems we can't access metadata fields so we convert to proto and back to do so? Hmm. Is this a pattern in the core repo too?